server/metadata.go

```diff
@@ -49,55 +60,68 @@ func (m *Metadata) readAll() error {
 		if !f.IsDir() {
 			continue
 		}
-		d, err := m.getAllMetadataFromFiles(f.Name())
+		v1Data, v2Data, err := m.getAllMetadataFromFiles(f.Name())
 		if err != nil {
 			return err
 		}
-		m.data[f.Name()] = &pbVersion.MetadataResponse{Versions: d}
+		if len(v1Data) > 0 {
+			m.v1Data[f.Name()] = &pbVersion.MetadataResponse{Versions: v1Data}
+		} else if len(v2Data) > 0 {
+			m.v2Data[f.Name()] = &pbVersion.MetadataV2Response{Versions: v2Data}
+		}
 	}
-
 	return nil
 }
 
-func (m *Metadata) getAllMetadataFromFiles(product string) ([]*pbVersion.MetadataVersion, error) {
+func (m *Metadata) getAllMetadataFromFiles(product string) ([]*pbVersion.MetadataVersion, []*pbVersion.MetadataV2Version, error) {
 	if !filepath.IsLocal(product) {
-		return nil, errors.New("product name is invalid")
+		return nil, nil, errors.New("product name is invalid")
 	}
 
 	dir := filepath.Join(".", product)
 	files, err := fs.ReadDir(m.fs, dir)
 	if err != nil {
-		return nil, errors.Join(err, errors.New("could not read metadata from directory"))
+		return nil, nil, errors.Join(err, errors.New("could not read metadata from directory"))
 	}
-
-	ret := make([]*pbVersion.MetadataVersion, 0, len(files))
+	v1Data := make([]*pbVersion.MetadataVersion, 0, len(files))
+	v2Data := make([]*pbVersion.MetadataV2Version, 0, len(files))
 	for _, f := range files {
 		p := filepath.Join(dir, f.Name())
 		c, err := fs.ReadFile(m.fs, p)
 		if err != nil {
-			return nil, errors.Join(err, fmt.Errorf("could not read file %s", p))
+			return nil, nil, errors.Join(err, fmt.Errorf("could not read file %s", p))
 		}
 
 		metaV, err := m.parseFile(c, filepath.Ext(f.Name()))
 		if err != nil {
-			return nil, errors.Join(err, fmt.Errorf("could not parse file %s", f.Name()))
+			return nil, nil, errors.Join(err, fmt.Errorf("could not parse file %s", f.Name()))
+		}
+		if metaV.ImageInfo == nil {
+			v1Data = append(v1Data, &pbVersion.MetadataVersion{
+				Version:     metaV.Version,
+				Recommended: metaV.Recommended,
+				Supported:   metaV.Supported,
+			})
```

Comment:
I think we need to add it to v2Data as well

Reply:
yeah that makes sense.

Reply:
Yeah that makes sense